bash-completion 2.9 breaks packages installing custom completions

**Describe the bug**
A clear and concise description of what the bug is.

dconf tries to write to the bash-completion store path.
```sh
glibPreInstallPhase
installing
install flags: install
[0/1] Installing files.
Traceback (most recent call last):
  File "/nix/store/8vaf7gf7ryzb2inm1zksncfchxc3w9kz-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/mesonmain.py", line 127, in run
    return options.run_func(options)
  File "/nix/store/8vaf7gf7ryzb2inm1zksncfchxc3w9kz-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/minstall.py", line 514, in run
    installer.do_install(datafilename)
  File "/nix/store/8vaf7gf7ryzb2inm1zksncfchxc3w9kz-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/minstall.py", line 346, in do_install
    self.install_data(d)
  File "/nix/store/8vaf7gf7ryzb2inm1zksncfchxc3w9kz-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/minstall.py", line 376, in install_data
    self.do_copyfile(fullfilename, outfilename)
  File "/nix/store/8vaf7gf7ryzb2inm1zksncfchxc3w9kz-meson-0.51.2/lib/python3.7/site-packages/mesonbuild/minstall.py", line 257, in do_copyfile
    shutil.copyfile(from_file, to_file)
  File "/nix/store/zdh16dcvjw99ybam59zd2ijb6bx138j0-python3-3.7.5/lib/python3.7/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
PermissionError: [Errno 13] Permission denied: '/nix/store/sqxcg0fssrc36kx8kfgmnrw52mkq4zb3-bash-completion-2.9/share/bash-completion/completions/dconf'
Installing service/dconf-service to /nix/store/0lmxp0r0973cg887pz2wnhgqipk6kpfw-dconf-0.34.0-lib/libexec
Installing gsettings/libdconfsettings.so to /nix/store/0lmxp0r0973cg887pz2wnhgqipk6kpfw-dconf-0.34.0-lib/lib/gio/modules
Installing client/libdconf.so.1.0.0 to /nix/store/0lmxp0r0973cg887pz2wnhgqipk6kpfw-dconf-0.34.0-lib/lib
Installing bin/dconf to /nix/store/clxwl76pphjw7d67f8f119r6ijyqj2kb-dconf-0.34.0/bin
Installing docs/dconf-service.1 to /nix/store/clxwl76pphjw7d67f8f119r6ijyqj2kb-dconf-0.34.0/share/man/man1
Installing docs/dconf.1 to /nix/store/clxwl76pphjw7d67f8f119r6ijyqj2kb-dconf-0.34.0/share/man/man1
Installing docs/dconf.7 to /nix/store/clxwl76pphjw7d67f8f119r6ijyqj2kb-dconf-0.34.0/share/man/man7
Installing /build/dconf-0.34.0/common/dconf-changeset.h to /nix/store/w14sl1abahj4msb5xrasrg7kpyip1xqw-dconf-0.34.0-dev/include/dconf/common
Installing /build/dconf-0.34.0/common/dconf-enums.h to /nix/store/w14sl1abahj4msb5xrasrg7kpyip1xqw-dconf-0.34.0-dev/include/dconf/common
Installing /build/dconf-0.34.0/common/dconf-paths.h to /nix/store/w14sl1abahj4msb5xrasrg7kpyip1xqw-dconf-0.34.0-dev/include/dconf/common
Installing /build/dconf-0.34.0/client/dconf.h to /nix/store/w14sl1abahj4msb5xrasrg7kpyip1xqw-dconf-0.34.0-dev/include/dconf
Installing /build/dconf-0.34.0/client/dconf-client.h to /nix/store/w14sl1abahj4msb5xrasrg7kpyip1xqw-dconf-0.34.0-dev/include/dconf/client
Installing /build/dconf-0.34.0/build/service/ca.desrt.dconf.service to /nix/store/clxwl76pphjw7d67f8f119r6ijyqj2kb-dconf-0.34.0/share/dbus-1/services
Installing /build/dconf-0.34.0/build/meson-private/dconf.pc to /nix/store/0lmxp0r0973cg887pz2wnhgqipk6kpfw-dconf-0.34.0-lib/lib/pkgconfig
Installing /build/dconf-0.34.0/client/dconf.vapi to /nix/store/clxwl76pphjw7d67f8f119r6ijyqj2kb-dconf-0.34.0/share/vala/vapi
Installing /build/dconf-0.34.0/client/dconf.deps to /nix/store/clxwl76pphjw7d67f8f119r6ijyqj2kb-dconf-0.34.0/share/vala/vapi
Installing /build/dconf-0.34.0/bin/completion/dconf to /nix/store/sqxcg0fssrc36kx8kfgmnrw52mkq4zb3-bash-completion-2.9/share/bash-completion/completions
FAILED: meson-install
/nix/store/8vaf7gf7ryzb2inm1zksncfchxc3w9kz-meson-0.51.2/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
builder for '/nix/store/3m79ll8z72w1zprgp9qcad2qvxyvb996-dconf-0.34.0.drv' failed with exit code 1
error: build of '/nix/store/3m79ll8z72w1zprgp9qcad2qvxyvb996-dconf-0.34.0.drv' failed
```
https://hydra.nixos.org/build/103951665

Note that bash-completions was updated in 12fd773ff058755237ebafe5f1c0bdd36f385eb6.

cc @jtojnar @jonringer 

**To Reproduce**
Steps to reproduce the behavior:
1. checkout staging
2. `nix-build -A gnome3.dconf`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: `gnome3.dconf`

--- a/pkgs/shells/bash/bash-completion/0001-Revert-build-Do-cmake-pc-and-profile-variable-replac.patch
+++ b/pkgs/shells/bash/bash-completion/0001-Revert-build-Do-cmake-pc-and-profile-variable-replac.patch
@@ -1,0 +1,86 @@
+From 398f44b4ed545fc1b6c13a057bf0900001f7958b Mon Sep 17 00:00:00 2001
+From: Frederik Rietdijk <fridh@fridh.nl>
+Date: Tue, 22 Oct 2019 15:07:05 +0200
+Subject: [PATCH] Revert "build: Do cmake, pc, and profile variable
+ replacements in Makefile"
+
+This reverts commit 81ba2c7e7dfbaefbafa1e8615727c9612e5fb314.
+---
+ Makefile.am                     | 12 ++++++++----
+ bash-completion-config.cmake.in |  4 ++--
+ bash-completion.pc.in           |  4 ++--
+ configure.ac                    |  3 +++
+ 4 files changed, 15 insertions(+), 8 deletions(-)
+
+diff --git a/Makefile.am b/Makefile.am
+index 8f441185..53979529 100644
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -23,8 +23,7 @@ cmakeconfig_DATA = bash-completion-config.cmake \
+ 		-e 's|@VERSION[@]|$(VERSION)|' \
+ 		<$(srcdir)/$@.in >$@
+ 
+-CLEANFILES = bash_completion.sh bash-completion.pc \
+-	bash-completion-config.cmake bash-completion-config-version.cmake
++CLEANFILES = bash_completion.sh bash-completion.pc
+ 
+ EXTRA_DIST = CHANGES $(pkgdata_DATA) bash_completion.sh.in .dir-locals.el \
+ 	.editorconfig README.md CONTRIBUTING.md pyproject.toml .perltidyrc \
+@@ -34,6 +33,11 @@ EXTRA_DIST = CHANGES $(pkgdata_DATA) bash_completion.sh.in .dir-locals.el \
+ install-data-hook:
+ 	tmpfile=`mktemp $${TMPDIR:-/tmp}/bash_completion.XXXXXX` && \
+ 	$(SED) -e 's|-/etc/bash_completion\.d|-$(compatdir)|' \
+-	    $(DESTDIR)$(pkgdatadir)/bash_completion >$$tmpfile && \
+-	cat $$tmpfile >$(DESTDIR)$(pkgdatadir)/bash_completion && \
++	    $(DESTDIR)$(pkgdatadir)/bash_completion > $$tmpfile && \
++	cat $$tmpfile > $(DESTDIR)$(pkgdatadir)/bash_completion && \
++	$(SED) -e 's|\$${prefix}|$(prefix)|' \
++	    $(DESTDIR)$(datadir)/cmake/$(PACKAGE)/bash-completion-config.cmake \
++	         > $$tmpfile && \
++	cat $$tmpfile > \
++	    $(DESTDIR)$(datadir)/cmake/$(PACKAGE)/bash-completion-config.cmake && \
+ 	rm $$tmpfile
+diff --git a/bash-completion-config.cmake.in b/bash-completion-config.cmake.in
+index d907b76c..ccc6e052 100644
+--- a/bash-completion-config.cmake.in
++++ b/bash-completion-config.cmake.in
+@@ -5,7 +5,7 @@ set (BASH_COMPLETION_VERSION "@VERSION@")
+ 
+ set (BASH_COMPLETION_PREFIX "@prefix@")
+ set (BASH_COMPLETION_COMPATDIR "@compatdir@")
+-set (BASH_COMPLETION_COMPLETIONSDIR "@pkgdatadir@/completions")
+-set (BASH_COMPLETION_HELPERSDIR "@pkgdatadir@/helpers")
++set (BASH_COMPLETION_COMPLETIONSDIR "@datarootdir@/@PACKAGE@/completions")
++set (BASH_COMPLETION_HELPERSDIR "@datarootdir@/@PACKAGE@/helpers")
+ 
+ set (BASH_COMPLETION_FOUND "TRUE")
+diff --git a/bash-completion.pc.in b/bash-completion.pc.in
+index ea03fd75..bde217db 100644
+--- a/bash-completion.pc.in
++++ b/bash-completion.pc.in
+@@ -1,7 +1,7 @@
+ prefix=@prefix@
+ compatdir=@compatdir@
+-completionsdir=@pkgdatadir@/completions
+-helpersdir=@pkgdatadir@/helpers
++completionsdir=@datarootdir@/@PACKAGE@/completions
++helpersdir=@datarootdir@/@PACKAGE@/helpers
+ 
+ Name: bash-completion
+ Description: programmable completion for the bash shell
+diff --git a/configure.ac b/configure.ac
+index 1f3b37e8..a216d9b7 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -13,5 +13,8 @@ helpers/Makefile
+ test/Makefile
+ test/t/Makefile
+ test/t/unit/Makefile
++bash-completion.pc
++bash-completion-config.cmake
++bash-completion-config-version.cmake
+ ])
+ AC_OUTPUT
+-- 
+2.21.0
+

--- a/pkgs/shells/bash/bash-completion/default.nix
+++ b/pkgs/shells/bash/bash-completion/default.nix
@@ -24,6 +24,10 @@ stdenv.mkDerivation rec {
     bashInteractive
   ];
 
+  patches = [
+    ./0001-Revert-build-Do-cmake-pc-and-profile-variable-replac.patch
+  ];
+
   # ignore ip_addresses because it tries to touch network
   # ignore test_ls because impure logic
   checkPhase = ''

